### PR TITLE
Adapt modal design to large screens

### LINKED ISSUE #2977: [Enhancement] Add desktop functionality to existing modal
- [x] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the enhancement
I would like to have the current modal (not ModalV2) enhanced with desktop styles and functionality during the migration period, so modal becomes a nice fallback until ModalV2 is feature complete. This includes: 

- [x] styling the modal wrapper component with correct paddings for header, footer and content
- [x] global styles that define width, height and border radius of modal on larger screens 
   - [x] multiple desktop sizes according to spec
- [x] removing drawer styles on desktop so it shows as normal modal 
- [x] show close button to the right instead of arrow down to the left in header on desktop
- [x] What should happen to the 'open in full height' option?
- [x] make modal scrollable to bottom on larger screen sizes
- [ ] update tests
- [x] modal shadow should dynamically scale 
- [x] min-height on fixed t-shirt sizes
- [ ] divider line in header
    - [x] always shown on larger screens 
    - [ ] should also be shown on mobile but only when modal is scrolled 
- [x] replace border radii with new 32px value 
- [x] make button in feature hint correct color

Known issues that needs fixing
- [ ] Keyboard pushes up modal and limits space - should keyboard appear on top?
- [ ] some modals cannot scroll all the way to the bottom, content is _cut off_ - it has something to do with _containment_ on modal wrapper
  ```
  @mixin contain-content() {
    ion-content {
        contain: content;
    }
  }
  ```

<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Refinement

- [ ] Request that the issue is [UX refined](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#ux-refinement); do not proceed until this is done.
- [ ] Request that the issue is [tech refined](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#tech-refinement); do not proceed until this is done.

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#before-you-get-coding)".
- [ ] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [develop branch](https://github.com/kirbydesign/designsystem/tree/develop) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Publish a WIP implementation to Github as a draft PR and ask for feedback. 
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Update the [cookbook](https://cookbook.kirby.design) with examples and showcases.

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request. If you created a draft PR during implementation you can just mark that as "ready for review".

### PR DESCRIPTION
**Pull request by RasmusKjeldgaard**
## Which issue does this PR close?

This PR closes #2977

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Reminders
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Make sure you have updated the cookbook with examples and showcases (for bug fixes, enhancements & new components).

### Review  
- [ ] Determine if your changes are a fix, feature or breaking-change, and add the matching label to your PR. If it is tooling, dependency updates or similar, add ignore-for-release.
- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review).
- [ ] Request that the changes are code-reviewed 
- [ ] Request that the changes are [UX reviewed](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#ux-review) (only necessary if your PR introduces visual changes)

When the pull request has been approved it will be merged to `develop` by Team Kirby.

### CODE REVIEW

**Comment by henrikvoetmand** [comment]
modal-drawer-review

File: libs/designsystem/modal/src/modal-wrapper/config/modal-config.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ export interface ModalConfig {
   collapseTitle?: boolean;
   component: any;
   size?: ModalSize;
+  height?: string;
```
Comment (RasmusKjeldgaard) [inline libs/designsystem/modal/src/modal-wrapper/config/modal-config.ts]:
```suggestion
  customHeight?: string;
```
Rename to `customHeight` as we want it to be clear that this is something that _overrides_ size

File: apps/cookbook/src/app/examples/example-configuration-wrapper/example-configuration-wrapper.component.scss [hunk 2]
```diff
@@ -1,7 +1,47 @@
...
+
+  &.snap-to-viewport {
+    kirby-card {
+      position: fixed;
+      margin-top: utils.$border-radius;
+      opacity: 0.5;
+      transition-property: transform, border-radius;
+      transition-duration: utils.get-transition-duration('short');
+      transition-timing-function: utils.get-transition-easing('enter-exit');
+      transform: rotate(-90deg) translateY(-#{utils.$border-radius + utils.size('xs')});
```
Comment (RasmusKjeldgaard) [inline apps/cookbook/src/app/examples/example-configuration-wrapper/example-configuration-wrapper.component.scss:57]: Restore previous functionality with no rotation and 40% transform on larger screens

File: apps/cookbook/src/app/examples/example-configuration-wrapper/example-configuration-wrapper.component.html [hunk 3]
```diff
@@ -1,4 +1,15 @@
-<ng-container *ngIf="snapToViewport; else fieldset">
+<button
```
Comment (RasmusKjeldgaard) [inline apps/cookbook/src/app/examples/example-configuration-wrapper/example-configuration-wrapper.component.html:1]: For line 20: `<legend>Configuration</legend>` should be configurable via input prop but default to 'Configuration'

File: apps/cookbook/src/app/examples/modal-example/embedded-modal-example/embedded-modal-example.component.html [hunk 4]
```diff
@@ -32,7 +32,9 @@ <h3>Open nested:</h3>
       <button kirby-button attentionLevel="2" (click)="showNestedActionSheet()">
         Action sheet
       </button>
-      <cookbook-example-configuration-wrapper>
+      <details>
+        <summary>Configuration for nested modal</summary>
+        <!-- <cookbook-example-configuration-wrapper [title]="'Configuration for nested modal'"> -->
```
Comment (RasmusKjeldgaard) [inline apps/cookbook/src/app/examples/modal-example/embedded-modal-example/embedded-modal-example.component.html]:
```suggestion

```

File: apps/cookbook/src/app/examples/modal-example/embedded-modal-example/embedded-modal-example.component.html [hunk 5]
```diff
@@ -32,7 +32,9 @@ <h3>Open nested:</h3>
       <button kirby-button attentionLevel="2" (click)="showNestedActionSheet()">
         Action sheet
       </button>
-      <cookbook-example-configuration-wrapper>
+      <details>
+        <summary>Configuration for nested modal</summary>
```
Comment (RasmusKjeldgaard) [inline apps/cookbook/src/app/examples/modal-example/embedded-modal-example/embedded-modal-example.component.html:36]: I love this one! 🙌

**Review by RasmusKjeldgaard — COMMENTED** [review]
Just reviewed the added docs stuff @jakobe, looks great!

**Review by jakobe — APPROVED** [review]
:shipit: 🤞🏻

**Pull request merged**